src/App.tsx

```diff
@@ -16,17 +19,27 @@ body {
 `;
 
 export default function App(): JSX.Element {
-  const { data: project, isLoading } = useFetch<ProjectDefinition>(GET_PROJECT_BY_ID(1));
+  const { data: projectResponse, isLoading } = useFetch<ProjectDefinition>(GET_PROJECT_BY_ID(1));
+  const [project, dispatch] = useReducer(projectReducer.reducer, projectReducer.initialState);
```

Comment:
This `useReducer`'s values are passed as context to children. You can use these values with `useTypeSafeContext` in any child components. These following are equivalent to redux state;
```typescript
  // this app
  const dispatch = useTypeSafeContext(ProjectReducerContext);

  // redux
  const dispatch = useDispatch();
```
```typescript
  // this app
  const project = useTypeSafeContext(ProjectContext);

  // redux
  const project = useSelector(projectSelector);
```